File: Guess Number Higher or Lower (374)/374.py
# Runtime: 20 ms
# Beats 94.44% of Python submissions

# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        left = 1
        right = n
        
        while left <= right:
            middle = (left + right) // 2
            
            if guess(middle) > 0:
                left = middle + 1
            elif guess(middle) < 0:
                right = middle - 1
            else:
                return middle

File: Guess Number Higher or Lower (374)/374.java
// Runtime: 0 ms
// Beats 100% of Java submissions

/* The guess API is defined in the parent class GuessGame.
   @param num, your guess
   @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
      int guess(int num); */

      public class Solution extends GuessGame {
        public int guessNumber(int n) {
            int left = 0;
            int right = n;
            
            while (left <= right) {
                int middle = left + ((right - left) / 2);
                int guess = guess(middle);
                
                if (guess == 0) {
                    return middle;
                } else if (guess == 1) {
                    left = middle + 1;
                } else {
                    right = middle - 1;
                }
            }
            return -1;
        }
    }